src/test/ui/union/union-derive-eq.stderr

```diff
@@ -1,10 +1,14 @@
 error[E0277]: the trait bound `PartialEqNotEq: std::cmp::Eq` is not satisfied
-  --> $DIR/union-derive-eq.rs:15:5
+  --> $DIR/union-derive-eq.rs:20:5
    |
 LL |     a: PartialEqNotEq,
    |     ^^^^^^^^^^^^^^^^^ the trait `std::cmp::Eq` is not implemented for `PartialEqNotEq`
+   | 
+  ::: $SRC_DIR/libcore/cmp.rs:LL:COL
+   |
+LL | pub struct AssertParamIsEq<T: Eq + ?Sized> {
+   |                               -- required by `std::cmp::AssertParamIsEq` here
    |
-   = note: required by `std::cmp::AssertParamIsEq`
```

Comment:
cc @nikomatsakis @Centril Finally a use for the per-`Predicate` `Span`s! I think I added those for the type alias work that never landed itself.

Reply:
Although for `AssertParamIsEq` specifically, we might want a custom `#[rustc_on_unimplemented]`, since it's used by `#[derive(Eq)]` specifically.

Reply:
Looking into this, `rustc_on_unimplemented` would need to have support for targeting bounds in the obligation chain too.